### DISCUSSION TITLE
q-scroll-area drag to scroll anywhere

### DISCUSSION THREAD

**Discussion by labs20** [quasarframework/quasar#14515] [General - Components / Directives / etc] [answered]
Hi. 

Is there some example around on how to make a scroll-area respond to "mousedrag" its contents by dragging anywhere inside of its area, and not just by dragging the scrollbars?

This could be a nice feat too, even more if I could lock scroll directions and define a css handler class to limit the scroll activation only on specific parts of my layout.

Thanks!

**Comment by pdanpdan** [comment, marked as answer]
You can check a description here: https://htmldom.dev/drag-to-scroll/
You can directly use VTouchPan - but the idea is bad all over:
- it's redundant on mobile (and it's best to only enable it on desktop)
- it breaks normal expectations for content (difficult text selection, you name it)

**Reply by labs20** [reply]
Hi, thanks!

Well, in my case I have a q-scroll-area with columns of cards, and each card has a top bar that the user should be able to grab and scroll the whole thing. Much like a carousel, but I have many columns visible at once.

Thanks for you link, in fact I was trying it before ask for help here =]

If someone have a more q-scroll-area comp oriented example, I'll be glad to read it too.

Thanks.

**Reply by pdanpdan** [reply]
if you start a codepen with scroll area and some cards (with a description of what should be active to scroll the area) we can help

**Reply by labs20** [reply]
I'll make one with an example of what I need. 

Thanks

**Reply by labs20** [reply]
Made this sketch: [https://codepen.io/labs20/pen/abGYpob](https://codepen.io/labs20/pen/abGYpob) 

Hope it helps to figure what I want.

Thanks in advance for your attention.

**Reply by pdanpdan** [reply]
https://codepen.io/pdanpdan/pen/VwxXpjd?editors=0010

All CSS changes are just cosmetic, so that I can see what I'm doing :)

**Reply by labs20** [reply]
v-touch-pan!!! Really nice!!

Thank you for your help. 

Would be nice to have such behaviour out of the box, though.

Appreciated.

**Comment by metalsadman** [comment]
https://codepen.io/metalsadman/pen/zYNEXwG

**Reply by labs20** [reply]
Thank you!
